code/game/objects/effects/spawners/random/maintenance.dm

```diff
@@ -10,17 +10,21 @@
 	. += span_info("This spawner has an effective loot count of [get_effective_lootcount()].")
 
 /obj/effect/spawner/random/maintenance/Initialize(mapload)
+	// Late loaded templates like shuttles can have maintenance loot.
+	// Once the game state progresses to roundstart, new maint loot spawners should just instantly pop.
+	if(SSticker.current_state >= GAME_STATE_SETTING_UP)
+		spawn_on_init = TRUE
+
 	. = ..()
-	// There is a single callback in SSmapping to spawn all delayed maintenance loot
-	// so we don't just make one callback per loot spawner
+
+	// If parent Initialize is calling for us to qdel, we can just return early.
+	if(. == INITIALIZE_HINT_QDEL || . == INITIALIZE_HINT_QDEL_FORCE)
+		return
+
+	// Otherwise we hang around until SSmapping spawns us in OnRoundstart and deletes us.
 	GLOB.maintenance_loot_spawners += src
 	loot = GLOB.maintenance_loot
```

Comment:
Will these things not be cleared on destroy anyway? why do this?

Reply:
They're used externally in SSmapping to spawn the loot OnRoundstart instead of on init.12

Reply:
`GLOB.maintenance_loot_spawners` is basically a cache of all maint loot spawners that don't fire on init.

![image](https://user-images.githubusercontent.com/24975989/134789637-9e0190f7-b310-4317-90b1-49c438adeb09.png)

The intent is that their map markers remain for players to see until `SSmapping.OnRoundstart()` when the maint loot is spawned and they're all deleted.

Reply:
it should be cleared in destroy anyway right? so why do the return check there?

Reply:
Needing to check for the hint sorta ignores the purpose of the hint in the first place.
The idea being that rather then qdeling in init, and then needing to QDELETING check after each parent call or risk hanging refs, we construct, and then immediately destruct, saving the headache.

This seems like it's recreating that pattern, and I think it's a bad standard to set for people

Reply:
@coiax who may have been part of that change IIRC